src/sensors/distant.cpp

```diff
@@ -44,21 +68,96 @@ distributed uniformly on the cross section of the scene's bounding sphere.
     bounding sphere cross section. Care should be taken notably when using the
     `constant` or `envmap` emitters.
 
-If the ``target`` parameter is set, the sensor looks at a single point and
+If the ``origin`` parameter is set, the sensor looks at a single point and
 records a (spectral) radiant flux per unit surface area per unit solid angle 
 (in unit power per unit surface area per unit solid angle).
 
+With the ``origin_`` parameters users can define an area parallel to the xy-plane
+that the sensor's rays will originate from. The area can be defined in two ways:
+
+- Setting `origin` to 'rectangle' lets users define a rectangular area through 
+  two points `origin_a` and `origin_b`. The resulting area will cover the area 
+  from the smaller to the larger value in both dimensions.
+- Setting `origin` to 'circle'  lets users define a circular origin area 
+  through a center `origin_center` and a radius `origin_radius`.
+- If `origin` is not set, the sensor will origin the entire cross section
+  of the scene's bounding sphere.
+
 */
 
-MTS_VARIANT class DistantSensor final : public Sensor<Float, Spectrum> {
+template <typename Float, typename Spectrum>
+class DistantSensor final : public Sensor<Float, Spectrum> {
+public:
+    MTS_IMPORT_BASE(Sensor, m_world_transform, m_film)
+    MTS_IMPORT_TYPES()
+
+    DistantSensor(const Properties &props) : Base(props), m_props(props) {
+
+        std::string origin;
```

Comment:
As commented last week, I think it would be clearer if this parameter (and associated variables) would be named `ray_origin`.